gobblin-core/src/main/java/gobblin/source/extractor/extract/sftp/SftpFsHelper.java

```diff
@@ -121,7 +165,7 @@ public InputStream getFileStream(String file)
       throws FileBasedHelperException {
     SftpGetMonitor monitor = new SftpGetMonitor();
     try {
-      return this.channelSftp.get(file, monitor);
+      return getSftpChannel().get(file, monitor);
```

Comment:
1. Channel does NOT get closed when inputStream is closed. when inputStream#close is called, jsch only sends a packet with close bytes to mark this inputStream as closed.
2. It does not seem like channel creation is expensive (from my basic code scan or JSCH.) Session creating is expensive as auth happens at session level.
Channels get cleaned up when session is closed. In MR mode, each mapper creates a session and a channel for ever file it opens. I do not think we expect thousands of files per mapper.
In this case is number of channels still a concern?